### PR TITLE
Reagents & Solutions

### PR DESCRIPTION
**Pull request by Acruid**
Adds a new Reagent yaml prototype for defining all your elements and chemicals. Adds a new Solution component to the ECS system for representing a container of a solution. A solution is a mixture of reagents. Visualize the Tinkers Construct furnace, that is what a solution of reagents looks like.

The Solution component also has some basic verbs for transferring solution between two containers. Each component has a set of capabilities for modeling the container. So you use the capabilities for modeling a syringe that can inject, or a bucket that can pour, etc.

For demo purposes a few chemicals and elements are added to play with. You can go ingame and transfer solution between containers. You can spawn a full water tank and take a bucket of water out of it.

Things that are not in this PR:
* Reactions between chemicals (TiCon alloying), this falls under the chemistry system.
* Injecting/Ingesting solutions, this falls under the health system.
* Any sort of world representation of solutions.

TLDR: This is a base system that other systems will depend on, but it does nothing much by itself.

### CODE REVIEW

File: Content.Shared/Chemistry/Solution.cs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        }
+
+        /// <summary>
+        ///     Adds a given quantity of a reagent directly into the solution. 
+        /// </summary>
+        /// <param name="reagentId">The prototype ID of the reagent to add.</param>
+        /// <param name="quantity">The quantity in milli-units.</param>
+        public void AddReagent(string reagentId, int quantity)
+        {
+            if(quantity <= 0)
```
Comment (PJB3005) [inline Content.Shared/Chemistry/Solution.cs]: Wouldn't it make more sense to throw an out of range exception if the value is negative?

File: Content.Shared/GameObjects/Components/Chemistry/SolutionComponent.cs [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Shared.GameObjects.Components.Chemistry
+{
+    public class SolutionComponent : Component
```
Comment (PJB3005) [inline Content.Shared/GameObjects/Components/Chemistry/SolutionComponent.cs]: For clarity it might be better to name it something like `SolutionContainer` isntead.

Comment (Acruid) [reply Content.Shared/GameObjects/Components/Chemistry/SolutionComponent.cs]: I avoided calling it a container because people might get the idea that it is only used for container-like entities (buckets, beakers). It will also be used for things like the human body blood, probably the oil/blood pools on the floor around the map, poisoned weapons, whatever liquids furry players need, etc.

The class name also follows the current naming convention of components, so idk if making this one a special snowflake would help.

File: Content.Shared/Chemistry/Solution.cs (hunk 1 above)
Comment (Acruid) [reply Content.Shared/Chemistry/Solution.cs]: dunno, the function as-is won't break if you give it a negative number. The exception would wreck the stack, and the only purpose would be to notify the client that they are trying to add a negative number. The code can handle a negative number, so is it really a program-stopping exception?
